lib/viewmodels/post_comment_count_view_model.dart

```diff
@@ -0,0 +1,35 @@
+import "package:hooks_riverpod/hooks_riverpod.dart";
+import "package:proxima/models/database/post/post_id_firestore.dart";
+import "package:proxima/services/database/post_repository_service.dart";
+
+/// This view model is used to keep in memory the number of comments of a post.
+/// It is refreshed every time the post comment list is refreshed, to stay consistent
+/// with it. It is also refreshed when a comment is deleted.
+/// The fact that it is auto-dispose allows to keep the memory usage low, keeping only
+/// the count of the comments of the posts that are currently displayed on the screen.
+/// This may moreover help mitigate some reloading bugs, since it suffices to scroll
+/// the post out of the screen to dispose the view model and thus the count of the comments.
+class PostCommentCountViewModel
+    extends AutoDisposeFamilyAsyncNotifier<int, PostIdFirestore> {
+  @override
+  Future<int> build(PostIdFirestore arg) async {
+    final postRepo = ref.watch(postRepositoryServiceProvider);
+    final post = await postRepo.getPost(arg);
+    return post.data.commentCount;
+  }
+
+  /// Set the count of comments of the post to [count]. This updates
+  /// the state without needing to fetch the data from the database.
+  void setCount(int count) {
+    state = AsyncValue.data(count);
+  }
+
+  Future<void> refresh() async {
+    state = await AsyncValue.guard(() => build(arg));
```

Comment:
Is it voluntary to have no loading state here ?

Refreshing a viewmodel is always done using the following:
```dart
  Future<void> refresh() async {
    state = const AsyncValue.loading();
    state = await AsyncValue.guard(() => build()); // with or without arg
  }
```

If yes why ?

If the goal here is to avoid the loading in the UI, that should be done directly in the UI, but the viewmodel should still have a loading state.

Reply:
It was voluntary. This came because I did not understand `AsyncValue::value` and `AsyncValue::valueOrNull` correctly. Now that I do, this was fixed in 810bf0977a466278e0921c81264e6ec3baea86e8. I moreover decided to use `value` instead of `valueOrNull` to propagate potential errors arrising in the `AsyncValue` in 70ac3f7c443984b6f99dba6c0578a805ac0d7b9a.

Reply:
Ok, I think I would have just kept `valueOrNull` instead of `value` to keep the code simpler and avoid user facing errors.